src/Network/Network/LoadBalancer/SetAzureLoadBalancerBackendPool.cs

```diff
@@ -0,0 +1,211 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Set, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "LoadBalancerBackendAddressPool"), OutputType(typeof(PSBackendAddressPool))]
+    public partial class SetAzureLoadBalancerBackendPool : NetworkBaseCmdlet
+    {
+        private const string SetByNameParameterSet = "SetByNameParameterSet";
+        private const string SetByParentObjectParameterSet = "SetByParentObjectParameterSet";
+        private const string SetByInputObjectParameterSet = "SetByInputObjectParameterSet";
+        private const string SetByResourceIdParameterSet = "SetByResourceIdParameterSet";
+
+        [Parameter(Mandatory = true, HelpMessage = "The resource group name of the load balancer.", ParameterSetName = SetByNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+
+        [Parameter(Mandatory = true,HelpMessage = "The name of the load balancer.",  ParameterSetName = SetByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string LoadBalancerName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the backend pool.", ParameterSetName = SetByNameParameterSet)]
+        [Parameter(Mandatory = true, HelpMessage = "The name of the backend pool.", ParameterSetName = SetByParentObjectParameterSet)]
```

Comment:
do we need this or we can get from inputObject.  What if they don't match?

Reply:
InputObject and Name is not in the same parameter set - user couldn't even call it and would get the below error. 

Set-AzLoadBalancerBackendAddressPool -InputObject $b2 -Name "fakeName"
Set-AzLoadBalancerBackendAddressPool : Parameter set cannot be resolved using the specified named parameters.


 Maybe you are referring to ParentObject here? That's the loadbalancer and name refers to backend pool here, its expected they won't match.